djangocms_versioning_filer/monkeypatch/admin/admin_fields.py

```diff
@@ -0,0 +1,184 @@
+        'admin_icon_delete': ('admin/img/icon-deletelink.svg'),
+    }
+    html = render_to_string('admin/filer/widgets/admin_file.html', context)
+    return mark_safe(html)
+
+
+def obj_for_value(self, value):
+    if value:
+        try:
+            key = self.rel.get_related_field().name
+            # we need the original manager to access draft
+            obj = self.rel.model._original_manager.get(**{key: value})
+        except Exception:
+            obj = None
+    else:
+        obj = None
+    return obj
+
+
+def __init__(self, rel, queryset, to_field_name, *args, **kwargs):
+    def remove_published_where(queryset):
+        '''We modify the queryset to generate choices that includes all files, even those in draft'''
+        where_children = queryset.query.where.children
+        all_except_published = [
+            lookup for lookup in where_children
+            if not (
+                lookup.lookup_name == 'exact' and
+                lookup.rhs == 'published' and
+                lookup.lhs.field.name == 'state'
+            )
+        ]
+
+        queryset.query.where = WhereNode()
+        queryset.query.where.children = all_except_published
+        return queryset
+
+    queryset = remove_published_where(queryset)
+    self.rel = rel
+    self.queryset = queryset
+    self.to_field_name = to_field_name
+    self.max_value = None
+    self.min_value = None
+    kwargs.pop('widget', None)
+    super(AdminFileFormField, self).__init__(queryset, widget=self.widget(rel, site), *args, **kwargs)
+
+
+def validate(self, value, model_instance):
+    from django.core import exceptions
+    from django.db import router
+    if self.remote_field.parent_link:
+        return
+    super(ForeignKey, self).validate(value, model_instance)
+    if value is None:
+        return
+
+    using = router.db_for_read(self.remote_field.model, instance=model_instance)
+    # we need to use the original manager to access drafts
+    qs = self.remote_field.model._original_manager.using(using).filter(
+        **{self.remote_field.field_name: value}
+    )
+    qs = qs.complex_filter(self.get_limit_choices_to())
+    if not qs.exists():
+        raise exceptions.ValidationError(
+            self.error_messages['invalid'],
+            code='invalid',
+            params={
+                'model': self.remote_field.model._meta.verbose_name, 'pk': value,
+                'field': self.remote_field.field_name, 'value': value,
+            },  # 'pk' is included for backwards compatibility
+        )
+
+
+filer.fields.file.AdminFileWidget.render = render
+filer.fields.file.AdminFileWidget.obj_for_value = obj_for_value
+filer.fields.file.AdminFileFormField.__init__ = __init__
+filer.fields.file.FilerFileField.validate = validate
+
+
+@receiver(post_version_operation)
+def post_version_publish_image_versions(sender, **kwargs):
+    '''Triggered when a page is published. This is when any image plugins also
+    should become published'''
```

Comment:
2 things here:

1. I don't agree with this: "Triggered when a page is published. This is when any image plugins also should become published". Moderation handles whether the contents should be published or not. In moderation the page may show draft items, add them to moderation (example one page, 2 files), some files may be rejected from moderation but the page can be published. With this code the files get published whether they were accepted in moderation or not.  

2. This is a signal not a monkey patch. It should live in signals.py although I'm not sure the signal is valid, moderation handles this scenario.

Reply:
@Aiky30 Yes, with this code the files get published when the page is accepted and published. The files that are in draft are completely sidestepping moderation in the code above. 

So if I understand you correctly, the behaviour we are looking for is that when a page is added to moderation collection, any images in draft mode should also be added to the same moderation collection. 

If moderation is not used at all the above behaviour is a valid behaviour.

Reply:
It's worth noting that if you have a file in draft that is the first version, and you end up publishing the page (and rejecting one draft file), that would file would presumably break on the published page. 

So that was one reason why I went with sidestepping any moderation here.

Reply:
I've removed this in its entirety as the outcome of the discussion that drafts needs to be submitted to moderation separately.